Feature: Remove object tags

dl3sdo opened the pull request:
Add dialog to remove object tags from selected objects that match a configurable pattern.
Allow to create an undo step.

Closes GH-2354 (Remove objects tags).

dl3sdo commented:
@dg0yt: I'm not sure whether using a 'trash' button in the tag editor widget is the right way for invoking the tag removal dialog. OTOH I did not find a suitable location in the menu and I don't want to overload Mapper's menu.

dg0yt reviewed (commented):
I think *you* want at least to move `TagRemoveDialog` to a different filename and to add it to the code check wrapper.

I'm not yet happy with the UI, but I can't propose actual improvements at the moment.

dl3sdo committed:
Remove object tags

Add dialog to remove object tags from selected objects using
a configurable pattern or a list of existing tags.
Allow to create an undo step.

Closes GH-2354 (Remove objects tags).

dl3sdo commented:
@dg0yt: Since I placed the action in the Map menu now should I move the .cpp and .h files to the gui/map folder (where e.g. map information and map notes are...but not the georeferencing dialog)?

dl3sdo commented:
There is currently a decoupling of the tags and objects being found (and shown) by the Find button and the actual deletion by the Remove button: If the user either changes the comparison operator or the key value and does not click Find again, the object tags that would be removed are not the ones actually being shown but those based on the current search parameters.

Proposal: Only remove those object tags that are the result of the search initiated by Find (and disable the Remove button if either the comparison operator or the key value field have changed).

dl3sdo commented:
/azp run

dl3sdo committed:
Add test for object tag removal
